packages/dapp/components/lib/hooks/useRouter.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import { AlphaRouter } from "@uniswap/smart-order-router";
+import { Percent, TradeType } from "@uniswap/sdk-core";
+import useWeb3 from "@/components/lib/hooks/useWeb3";
+import { uAD, uCR, USDC, USDT, DAI, parseAmount } from "../utils";
+
+const useRouter = (selectedToken: string, amountIn = "0"): string | undefined => {
+  const [{ provider, walletAddress }] = useWeb3();
+  const [quotePrice, setQuotePrice] = useState<string>();
+
+  async function getQuote() {
```

Comment:
It's unfortunate because it seems like you put in a fairly significant amount of work on implementing the router. But if you think that the widget solves the business case (I think it does last I recall our conversation on it) then this half implemented router solution related code should all be removed unfortunately.